pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -132,12 +133,21 @@ func (h *appHandler) apply(ctx context.Context, appRev *v1beta1.ApplicationRevis
 		return h.createOrUpdateAppRevision(ctx, appRev)
 	}
 
+	var needTracker bool
+	var err error
 	for _, comp := range comps {
 		comp.SetOwnerReferences(owners)
-		needTracker, err := h.checkAndSetResourceTracker(&comp.Spec.Workload)
+		if h.checkAutoDetect(comp) && h.isNewRevision {
```

Comment:
why it's related with `h.isNewRevision` ?

Reply:
please clarify in the code

Reply:
sorry, i will fix it